src/webui/src/components/Modal/CustomizedTrial.tsx

```diff
@@ -0,0 +1,317 @@
+import * as React from 'react';
+import axios from 'axios';
+import { Row, Col, Input, Modal, Form, Button, Icon } from 'antd';
+import { MANAGER_IP } from '../../static/const';
+import { EXPERIMENT, TRIALS } from '../../static/datamodel';
+import { FormComponentProps } from 'antd/lib/form';
+const FormItem = Form.Item;
+import './customized.scss';
+
+interface CustomizeProps extends FormComponentProps {
+    visible: boolean;
+    copyTrialId: string;
+    closeCustomizeModal: () => void;
+}
+
+interface CustomizeState {
+    isShowSubmitSucceed: boolean;
+    isShowSubmitFailed: boolean;
+    isShowWarning: boolean;
+    searchSpace: object;
+    copyTrialParameter: object; // user click the trial's parameters
+    customParameters: object; // customized trial, maybe user change trial's parameters
+    customID: number; // submit customized trial succeed, return the new customized trial id
+}
+
+class Customize extends React.Component<CustomizeProps, CustomizeState> {
+
+    constructor(props: CustomizeProps) {
+        super(props);
+        this.state = {
+            isShowSubmitSucceed: false,
+            isShowSubmitFailed: false,
+            isShowWarning: false,
+            searchSpace: EXPERIMENT.searchSpace,
+            copyTrialParameter: {},
+            customParameters: {},
+            customID: NaN
+        };
+    }
+
+    // [submit click] user add a new trial [submit a trial]
+    addNewTrial = () => {
+        const { searchSpace, copyTrialParameter } = this.state;
+        // get user edited hyperParameter, ps: will change data type if you modify the input val
+        const customized = this.props.form.getFieldsValue();
+        // true: parameters are wrong
+        let flag = false;
+        Object.keys(customized).map(item => {
+            if (item !== 'tag') {
+                // unified data type
+                if (typeof copyTrialParameter[item] === 'number' && typeof customized[item] === 'string') {
+                    customized[item] = JSON.parse(customized[item]);
+                }
+                if (searchSpace[item]._type === 'choice') {
+                    if (searchSpace[item]._value.find((val: string | number) =>
+                        val === customized[item]) === undefined) {
+                        flag = true;
+                        return;
+                    }
+                } else {
+                    if (customized[item] < searchSpace[item]._value[0]
+                        || customized[item] > searchSpace[item]._value[1]) {
+                        flag = true;
+                        return;
+                    }
+                }
+            }
+        });
+        if (flag !== false) {
+            // open the warning modal
+            this.setState(() => ({ isShowWarning: true, customParameters: customized }));
```

Comment:
`() =>` is unnecessary in all `setState`s.
`this.setState(() => ({ k: v }));` is effectively identical to `this.setState({ k: v });`.